common/ops/gluster_ops/snapshot_ops.py

```
@@ -151,3 +152,496 @@ def is_snapd_running(self, volname: str, node: str) -> bool:
+        cmd = (f"gluster snapshot clone {clonename} {snapname} --mode=script"
+               " --xml")
+        return self.execute_abstract_op_node(cmd, node, excep)
+
+    def snap_restore(self, snapname: str, node: str,
+                     excep: bool = True) -> bool:
+        """
+        Method to restore the snapshot.
+
+        Args:
+            snapname (str): Name of the snapshot
+            node (str): Node wherien the command is to be executed.
+
+        Optional:
+            excep (bool): Flag to control exception handling by the
+            abstract ops. If True, the exception is handled, or else
+            it isn't.
+
+        Returns:
+            ret: A dictionary consisting
+                - Flag : Flag to check if connection failed
+                - msg : message
+                - error_msg: error message
+                - error_code: error code returned
+                - cmd : command that got executed
+                - node : node on which the command got executed
+        """
+        cmd = (f"gluster snapshot restore {snapname} --mode=script --xml")
+        return self.execute_abstract_op_node(cmd, node, excep)
+
+    def snap_restore_complete(self, volname: str, snapname: str,
+                              node: str, excep: bool = True) -> bool:
+        """
+        Method restore the snapshot and that is done when volume is stopped.
+        Post snap restore, the volume is started again.
+
+        Args:
+            volname (str): Name of the volume.
+            snpaname (str): name of the snapshot.
+            node (str): node wherien the command is to be executed.
+
+        Optional:
+            excep (bool): Flag to control exception handling by the
+            abstract ops. If True, the exception is handled, or else
+            it isn't.
+
+        Returns:
+            bool: True if restore is a success or False.
+        """
+        # Stop the volume.
+        self.volume_stop(volname, node, force=True)
```

Is it necessary for volume to be stopped using force?

---

If not then we should do a fail check here as well, by passing `excep=False` by default inside this function.

---

Well, the focus here is on the snap_restore when volume is stopped. Hence it is better to use force. Now we can make this configurable of whether to give force option or not. But seeing the test cases, I don't see a need to add one more param. We can modify on future if it needs to be configurable.

---

I mean functionality-wise we shouldn't be doing a stop and start with force, that way we can bypass few errors in the code. Better to check the return value of the stop and start function call by passing `excep=False` to the function and return `False` from the caller i.e, `snashot_restore_complete` in case of failure.

---

Cool, we can remove the force, but for excep, we should be taking the value user has passed. If the user doesn't want a return value specially if things fail, we can let it fail in case of volume start/stop itself.